x-pack/plugins/lens/public/datasources/form_based/to_expression.ts

```diff
@@ -142,7 +142,7 @@ function getExpressionForLayer(
       if (def.input !== 'fullReference' && def.input !== 'managedReference') {
         const aggId = String(index);
 
-        const wrapInFilter = Boolean(def.filterable && col.filter);
+        const wrapInFilter = Boolean(def.filterable && col.filter?.query);
```

Comment:
This change prevents us wrapping this agg in a filter when the query is just an empty string. No difference on the elasticsearch query, but this does make sure that if someone removes a kql filter from a dimension, that agg can be collapsed into other matching aggs instead of being passed over because it looks like it has a query.